src/backend/integrations/builtin/obs/obs-remote.ts

```diff
@@ -785,6 +785,99 @@ export async function getAudioSources(): Promise<Array<OBSSource>> {
     return audioSupportedSources;
 }
 
+export async function getTransformableSceneItems(sceneName: string): Promise<Array<OBSSceneItem>> {
+    const sceneItems = await getAllSceneItemsInScene(sceneName) ?? [];
+    const sources = (await getAllSources()) ?? [];
+
+    return sceneItems.filter(item => sources.some(source => source.name === item.name && !source.typeId.startsWith("wasapi")));
+}
+
+function getLerpedCallsArray(sceneName: string, sceneItemId: number, transformStart: Record<string, number>, transformEnd: Record<string, number>, duration: number, easeIn = false, easeOut = false) {
+    if (!transformEnd || !Object.keys(transformEnd).length) {
+        return [];
+    }
+
+    const calls = [];
+    const interval = 1 / 60;
+
+    calls.push({
+        requestType: "SetSceneItemTransform",
+        requestData: {
+            sceneName,
+            sceneItemId,
+            sceneItemTransform: transformStart
+        }
+    });
+
+    let time = 0;
+    let first = true; // Ensure end value is used even if duration is 0
+    while (first || time < duration) {
```

Comment:
A `do...while` loop could be utilized here with the same effect, as the "do" block is executed before the first while check:
```ts
let time = 0;
do {
 //...
} while (time < duration);
```

Reply:
I always forget this syntax exists, it's so niche, but 100% perfect for this

Reply:
Done!